docs/proposals/MultiSelect/MultiSelect.md

```diff
@@ -0,0 +1,105 @@
+# `MultiSelect`
+
+The `MultiSelect` component gives our users the possibility to select multiple
+options from a menu.
+
+## Design Patterns
+
+The goal of the `MultiSelect` component is to allow a user to check multiple
+options within a list of items.
+
+A user can click on the element to expand it and display the multiple options.
+The element will display the selected options or the placeholder text if none
+are selected.
+
+Every time an option is checked or unchecked a behavior can be triggered using
+the onChange prop
+
+### How it might behave:
+
+- This multi-select dropdown matches the designs seen in the Figma here
+
+- Developer should have props for whether or not an “All” checkbox should show
+  up at the top, as well as the text that should display in this option
+
+  - The “All” checkbox should be indeterminate
+
+    - If it’s currently unchecked and the user checks it then all of the lower
+      options should be checked automatically
+
+      - If unchecked again after that then all of the lower options should be
+        unchecked automatically
+
+    - The “All” checkbox is automatically marked checked if the user manually
+      checks all of the options below it
+
+    - The “All” checkbox is automatically marked unchecked if the user manually
+      unchecks all of the options below it
+
+- The text that appears in the box before a user clicks on it should be either
+  “All **\_\_\_\_**” or a comma separated list of the current selections
+
+## Accessibility
```

Comment:
👌

Reply:
Love these considerations. One thing I'm wondering is whether we can simply use an `html` select ([with multiple enabled](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select#attr-multiple)) internally, as it should already have most of the accessibility we're looking for?

Reply:
Hi Chris, the default `html select` only supports the `option html object` as child elements, so it is not possible to implement the requested design that required `checkboxes`.

Reply:
I'm just referencing the accessibility-facing side of things. We do similar things with our radio button, for example, where the actual element the user is interacting with is different from the visual element represented on-screen because of styling limitations with the underlying `html` element:
![image](https://user-images.githubusercontent.com/39704901/194604876-34aa3a06-a1b2-454f-93b6-a2a72783ef09.png)

Reply:
In this example, the radio input is actually rendered offscreen, but is still present in the DOM - so the user gets the same accessibility, while we have some more control over the visual presentation of the element.

Reply:
I was taking a look at the RadioGroup component and I understood it simulates the radio behavior though css in the real component that the user is interacting with. I did some tests and it is possible to do the same with the default select and options elements if the selectable options were the only thing within the options container. 

But it seems like more will have to be implemented in the future on this component, this is a screenshot of the Finsights project that have more elements in the options container like search bar and reset button. 

![image](https://user-images.githubusercontent.com/27180625/194881148-6d338466-65ad-42b6-bb5b-60f56db9dc8e.png)

For that reason I believe it is not possible to implement the default select with the default options elements with this design, so the options component all of the accessibility will have to be implemented from scratch.

Reply:
Got it, thanks @CHPrado

Reply:
@chris-at-jobber I added the accessibility requirements you suggested.